services/document-manager/backend/app/docman/utils/document_upload_helper.py

```diff
@@ -285,3 +293,153 @@ def complete_upload(cls, key, new_key, doc_guid, versions=None, version_guid=Non
+
+    @classmethod
+    def validate_bundle(cls, bundle_documents):
+        required_extensions = {'.shp', '.shx', '.dbf', '.prj'}
+        optional_extensions = {'.sbn', '.sbx', '.xml'}
+        all_shape_file_extensions = required_extensions.union(optional_extensions)
+        valid_single_file_extensions = {'.kml', '.kmz'}
+
+        if len(bundle_documents) == 1:
+            document_extension = os.path.splitext(bundle_documents[0].file_display_name)[1]
+
+            if document_extension in all_shape_file_extensions:
+                raise ValueError(f'${document_extension} must be uploaded as part of a shapefile')
+            if document_extension not in valid_single_file_extensions:
+                raise ValueError(f'Invalid file type: {document_extension}')
+        else:
+            allowed_extensions = required_extensions.union(optional_extensions)
+
+            # assuming Document.file_display_name or similar method gives file name including extension
+            document_extensions = {os.path.splitext(doc.file_display_name)[1] for doc in bundle_documents}
+
+            if not required_extensions.issubset(document_extensions):
+                missing_extensions = required_extensions - document_extensions
+                raise ValueError(f'Missing required file types: {", ".join(missing_extensions)}')
+
+            extra_extensions = document_extensions - allowed_extensions
+            if extra_extensions:
+                raise ValueError(
+                    f'Found non spatial bundle file types in spatial bundle: {", ".join(extra_extensions)}')
+
+    @classmethod
+    def zip_spatial_files(cls, bundle_documents, name):
+        oss_service = ObjectStoreStorageService()
+        shpz_path = secure_filename(f'{name}.shpz')
+
+        with zipfile.ZipFile(shpz_path, 'w', zipfile.ZIP_DEFLATED) as zipf:
+            for doc in bundle_documents:
+                response = oss_service.download_file(doc.object_store_path, doc.file_display_name, False)
+
+                if response.status_code == 200:
+                    current_app.logger.info(f"Successfully downloaded document: {doc.file_display_name}")
+                    file_data = response.get_data()
+
+                    zipf.writestr(doc.file_display_name, file_data)
+                else:
+                    raise Exception(f"Failed to download document: {doc.file_display_name}")
+
+        return shpz_path
+
+    @classmethod
+    def download_kml_kmz_files(cls, doc):
+        # Download the file from object store
+        response = ObjectStoreStorageService().download_file(doc.object_store_path, doc.file_display_name, False)
+
+        # Check the response status
+        if response.status_code == 200:
+            file_data = response.get_data()  # obtain file data from response body
+
+            dir_path = 'tmp/spatial'
+            os.makedirs(dir_path, exist_ok=True)
+            file_path = os.path.join(dir_path, doc.file_display_name)
```

Comment:
Could we also do the `secure_filename` on this one? 
Just good practice to do with any user provided names of files to avoid any chance of path traversal attacks